app/services/service_listeners/publishing_api_associated_documents.rb

```diff
@@ -92,10 +115,19 @@ def locale_for_document(document)
 
     def discard_drafts(associated_documents)
       associated_documents.each do |associated_document|
-        PublishingApiDiscardDraftWorker.perform_async(
-          associated_document.content_id,
-          edition.primary_locale,
-        )
+        if associated_document.respond_to?(:translations)
+          associated_document.translations.each do |translation|
+            PublishingApiDiscardDraftWorker.perform_async(
+              associated_document.content_id,
+              translation.locale.to_s,
+            )
+          end
+        else
+          PublishingApiDiscardDraftWorker.perform_async(
+            associated_document.content_id,
+            locale_for_document(associated_document).to_s,
+          )
+        end
```

Comment:
Would it be possible to add tests for the `unpublish`, `discard_drafts` and `withdraw` scenarios where the associated documents have translations?  I don't think that's currently covered.